build/create-jira-release

```diff
@@ -0,0 +1,126 @@
+const OAuth = require('oauth').OAuth
+
+const JIRA_URL = 'https://issues.integreat-app.de'
+
+program
+  .option('-d, --debug', 'enable extreme logging')
+  .requiredOption('--project-name <project-name>', 'the name of the jira project, e.g. integreat-webapp')
+  .requiredOption('--access-token <access-token>', 'version name of the new release')
+  .requiredOption('--private-key <privateKey>')
+  .requiredOption('--consumer-key <consumer-key>')
+
+const initializeOAuth = (privateKeyBase64, consumerKey) => {
+  const privateKey = Buffer.from(privateKeyBase64, 'base64').toString('ascii')
+
+  return new OAuth(
+    '',
+    '',
+    consumerKey,
+    privateKey,
+    '1.0',
+    '',
+    'RSA-SHA1'
+  )
+}
+
+const postToJira = async (path, body, accessToken, privateKeyBase64, consumerKey) => {
+  const consumer = initializeOAuth(privateKeyBase64, consumerKey)
+
+  await new Promise((resolve, reject) => consumer.post(
+    `${JIRA_URL}${path}`,
+    accessToken,
+    '',
+    JSON.stringify(body),
+    'application/json',
+    (error, response) => error ? reject(error) : resolve(response)
+  ))
+}
+
+const putToJira = async (path, body, accessToken, privateKeyBase64, consumerKey) => {
+  const consumer = initializeOAuth(privateKeyBase64, consumerKey)
+
+  await new Promise((resolve, reject) => consumer.put(
+    `${JIRA_URL}${path}`,
+    accessToken,
+    '',
+    JSON.stringify(body),
+    'application/json',
+    (error, response) => error ? reject(error) : resolve(response)
+  ))
+}
+
+const getFromJira = async path => {
+  const response = await fetch(`${JIRA_URL}${path}`)
+  return response.json()
+}
+
+const createRelease = async ({ newVersionName, accessToken, privateKey, consumerKey, projectName }) => {
+  const date = new Date()
+  const newReleaseDate = `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`
+
+  const projects = await getFromJira('/rest/api/2/project')
+  const project = projects.find(project => project.name === projectName)
+  const projectKey = project.key
+
+  // Create a new jira version
+  await postToJira('/rest/api/2/version', {
+    name: newVersionName,
+    released: true,
+    releaseDate: newReleaseDate,
+    project: projectKey
+  }, accessToken, privateKey, consumerKey)
+
+  // Get the new and the last release
+  const { values: versions } = await getFromJira(
+    `/rest/api/2/project/${projectKey}/version?orderBy=-releaseDate&maxResults=2`
+  )
+
+  const lastVersion = versions[1]
+  const lastDate = lastVersion.releaseDate
+  const newVersion = versions[0]
+
+  if (newVersion.name !== newVersionName) {
+    throw Error(`There is already a release ${newVersion.name} with a later release date in the project!`)
+  }
+
+  // Get all issues which should be part of the new release
+  const query =
+    `project = ${projectName} AND fixVersion IS EMPTY AND resolution = Done AND resolutiondate > ${lastDate}`
+  const { issues } = await getFromJira(`/rest/api/2/search?jql=${query}&fields=id`)
```

Comment:
this returns maximum of 50 issues. We should either paginate here or set maxResults to a higher number. Even though I am not sure of how many issues we are talking about here normally. ^^

Reply:
Hmm in theory we do a release every twice a month, I don't think we'll ever make it to resolve more than 50 issues in that time :D

Reply:
Increased to 1000 issues. If this is not enough, I guess we have other problems :sunglasses: